pyopia/instrument/silcam.py

```diff
@@ -23,41 +25,63 @@ def timestamp_from_filename(filename):
     return timestamp
 
 
+@dataclass
+class LoadData:
+    timestamp: Any
+    img: Any
+
+
 class SilCamLoad():
     '''PyOpia pipline-compatible class for loading a single silcam image
 
+    Requires pipeline data: :class:`LoadData`
+
     Parameters
     ----------
     filename : string
         silcam filename (.silc)
 
     Returns
     -------
-    timestamp : timestamp
-        timestamp from timestamp_from_filename()
-    img : np.array
-        raw silcam image
+        data.img : :class:`LoadData`
+        data.timestamp : :class:`LoadData`
     '''
 
     def __init__(self, filename):
         self.filename = filename
 
-    def __call__(self, data):
+    def __call__(self, data: LoadData) -> LoadData:
         timestamp = timestamp_from_filename(self.filename)
         img = np.load(self.filename, allow_pickle=False)
         data['timestamp'] = timestamp
         data['img'] = img
         return data
 
 
+@dataclass
+class ImagePrepData(LoadData):
+    img: np.ndarray
+    imc: np.ndarray
+
+
 class ImagePrep():
+    '''Simplify processing by squeezing a 3-channel image into a 2D array
+
+    min is used for squeezing to represent the highest attenuation of all wavelengths
+
+    Requires pipeline data: :class:`ImagePrepData`
+
+    Returns
+    -------
+        data.imc : :class:`ImagePrepData`
+    '''
 
     def __init__(self):
         pass
 
-    def __call__(self, data):
+    def __call__(self, data: ImagePrepData) -> ImagePrepData:
```

Comment:
I think this should take LoadData as input and only return ImagePrepData. The difference between them is imc, and that is created here